### DISCUSSION TITLE
Do you need to change your slicer settings after installing firmware?

### DISCUSSION THREAD

**Discussion by discopandan** [synman/Ender-3-S1-Pro-Firmware#58] [Q&A]
I'm new to 3D printing so not sure how things are handled, but I know I'm annoyed with the stock firmware. 

For example it is mentioned in the capabilities list that: 

Probe X/Y offset updated to be more accurate
Bed Size set to 235mm x 225mm (S1 Pro)
Or the different bed levelling methods

Is that something that needs to be taken into account in the slicer?

**Comment by ghylander** [comment]
I'm interested about this as well.

In addition to the questions raised by OP:
Currently, stock firmware restores bed mesh after G28. Is this setting still enabled?
Do I have to mdify the start gcode to manually probe new mesh/restore saved mesh?

**Reply by jklap** [reply]
`RESTORE_LEVELING_AFTER_G28` is set in @synman's firmware

But your comment is "manually probe new mesh/restore saved mesh" -- if you want to probe a new mesh then yes, you will need to add that to your start gcode. If you just want to restore/activate the saved mesh, then no. But you can always just add `M420 S` and not worry about it!

**Reply by ghylander** [reply]
Great, thanks!

The comment about probe new mesh before printing was about `RESTORE_LEVELING_AFTER_G28`. Coming from a CR20, this printer was very confusing to set up and level.

I had to search online for instructions because the manual povided by creality didn't really clear things up. Then I had similar trouble with the slicers (I use PrusaSlicer). It seems that if the printer FW has `RESTORE_LEVELING_AFTER_G28` and you use G29 in the gcode, the printer errors out.

My comment meant "if `RESTORE_LEVELING_AFTER_G28` is not enabled in the FW, do I have to modify the start gcode to probe a new mesh or restore the saved mesh?"
